frontend/src/layout/navigation/TopBar/announcementLogic.ts

```diff
@@ -119,5 +124,14 @@ export const announcementLogic = kea<announcementLogicType>({
                     : null
             },
         ],
-    },
-})
+    }),
+    urlToAction(({ values, actions }) => ({
+        '*': ({ pathname }) => {
+            if (values.canShowAnnouncements && pathname?.startsWith('/ingestion')) {
+                actions.setCanShowAnnouncements(false)
+            } else if (!values.canShowAnnouncements) {
+                actions.setCanShowAnnouncements(true)
+            }
```

Comment:
Late to the party, but a suggestion for the future. I find that it's often more practical to name boolean actions separately like `showAnnouncements` and `hideAnnouncements`, instead of `setX`.

...

However, I'm not sure this is the cleanest way to keep track of this. Usually, if it's a value you can somehow derive as a selector, you should derive it as a selector... instead of having code that listens for things, checks other things and then sets another set of things. 

In this case, you could have a selector like:

```ts
selectors({
  canShowAnnouncements: [
    () => [router.selectors.location], ({ pathname }) => pathname.startsWith('/ingestion')
  ]
})
```

or even better, avoid hardcoded pathnames and use `urls.ingestion()` :).